### PR TITLE
[refactor] Delete RunRecord.pipeline_run

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Delete `RunRecord.pipeline_run` (update refs to `RunRecord.dagster_run`)

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12368** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12368" target="_blank">Graphite</a> 
    * **PR #12369** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12369" target="_blank">Graphite</a> 
      * **PR #12370** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12370" target="_blank">Graphite</a> 
        * **PR #12371** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12371" target="_blank">Graphite</a> 
          * **PR #12372** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12372" target="_blank">Graphite</a> 
            * **PR #12380** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12380" target="_blank">Graphite</a> 
            * **PR #12373** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12373" target="_blank">Graphite</a> 
              * **PR #12374** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12374" target="_blank">Graphite</a> 
                * **PR #12331** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12331" target="_blank">Graphite</a> 
                * **PR #12383** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12383" target="_blank">Graphite</a> 
                  * **PR #12384** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12384" target="_blank">Graphite</a> 
                    * **PR #12385** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12385" target="_blank">Graphite</a>  👈
                      * **PR #12386** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12386" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12385?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/storage/pipeline_run.py [hunk 1]
```diff
@@ -713,18 +713,14 @@ def __new__(
         return super(RunRecord, cls).__new__(
             cls,
             storage_id=check.int_param(storage_id, "storage_id"),
-            pipeline_run=check.inst_param(pipeline_run, "pipeline_run", DagsterRun),
+            dagster_run=check.inst_param(dagster_run, "pipeline_run", DagsterRun),
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/storage/pipeline_run.py:716]: fix `"pipeline_run"`

**Review by alangenfeld — APPROVED** [review]
not whitelisted for serdes, you love to see it

**Comment by alangenfeld** [comment]
since `RunRecord` is top level i think you will still want to get a changelog entry for this change to be safe - or just keep around a getter for `pipeline_run`

**Comment by alangenfeld** [comment]
i would advocate for keeping a getter around since its so little code / burden

**Comment by smackesey** [comment]
>i would advocate for keeping a getter around since its so little code / burden

OK, I added a getter that we can drop for 1.2

**Pull request merged**